### ISSUE TITLE
Build failing

### ISSUE THREAD

**Issue by bsvdoom** [OrangePiLibra/OrangePi_H5SDK#9]
Indeed its not a dir, its the mali driver module binary..
`-rw-r--r-- 1 user user 5896110 dec    8 19:08 /home/user/OrangePi_H5SDK/output/lib
`
```
'mali450/kernel_mode/driver/src/devicedrv/mali/mali.ko' -> '/home/user/OrangePi_H5SDK/output/lib'
 Build Mali450 succeed
 Start Install Module
make: Entering directory '/home/user/OrangePi_H5SDK/kernel'
mkdir: cant create directory: ”/home/user/OrangePi_H5SDK/output/lib”: Not a directory 
Makefile:970: recipe for target '_modinst_' failed
make: *** [_modinst_] Error 1
make: Leaving directory '/home/user/OrangePi_H5SDK/kernel'
```

**Comment by sv1eia** [comment]
@BuddyZhang1 
Yes, tested here and build fails after the latest commit with the given mali patch .
-> [c4cb7a0](https://github.com/OrangePiLibra/OrangePi_H5SDK/commit/c4cb7a013ee037a34073d8fd98a18d4a9aca904a)

Tried a full image xenial build and received the same as @bsvdoom

**Comment by jernejsk** [comment]
Do you have user space mali450 library? X11 and fbdev version?

**Comment by BuddyZhang1** [comment]
@jernejsk 
Now, I porting mali450 into OrangePi H5 with Allwinner engineer.
I will release lib of Mali450 user space driver.

**Comment by BuddyZhang1** [comment]
@bsvdoom 
  Pls build linux kernel first.

**Comment by bsvdoom** [comment]
@BuddyZhang1 
I am trying, but with the new commit ( 94b8798 ) comes new error:

Conver script to dts ok.
./pack: line: 185: /home/user/OrangePi_H5SDK/scripts/pack/pctools/linux/fsbuild200/fsbuild: No such file or directory
cp: stat 'boot-resource.fex' unsuccesful: No such file or directory
pack boot package
GetPrivateProfileSection read to end
content_count=3
file /home/user/OrangePi_H5SDK/output/pack/out/u-boot.fex cant be open
create package failed
ERROR: dragonsecboot  boot_package.cfg failed

**Comment by sv1eia** [comment]
@BuddyZhang1 
Performed a new git pull and then a new build for full image xenial.
Got again a build failure.

**Comment by ehoutsma** [comment]
I'm missing the source/link for bl31.bin (arm-trusted-firmware?) and scp.bin so we can build and boot from this source, that would be awesome!

**Comment by sv1eia** [comment]
@BuddyZhang1 

Did a new git clone, the latest commit is
-> [commit 94b8798](https://github.com/OrangePiLibra/OrangePi_H5SDK/commit/94b879894483fc56cfc847269118fc8b1dd3a6d8)

Tried a full image xenial build
The error got is

********************************
Conver script to dts ok.
pack boot package
GetPrivateProfileSection read to end
content_count=3
file /home/user/orangepi/OrangePi_H5SDK/output/pack/out/u-boot.fex cant be open
create package failed
ERROR: dragonsecboot  boot_package.cfg failed
********************************

**Comment by sv1eia** [comment]
@BuddyZhang1 

Also when trying to build only uboot, these errors come

*******************************************
Compile U-boot......
Configuring for sun50iw2p1 - Board: sun50iw2p1, Options: SUN50IW2P1
make: /home/user/orangepi/OrangePi_H5SDK/u-boot/../toolchain/gcc-linaro-aarch/gcc-linaro/bin/arm-linux-gnueabi-gcc: Command not found
/bin/sh: 1: /home/user/orangepi/OrangePi_H5SDK/u-boot/../toolchain/gcc-linaro-aarch/gcc-linaro/bin/arm-linux-gnueabi-gcc: not found
dirname: missing operand
Try 'dirname --help' for more information.
Makefile:1203: *** missing separator.  Stop.
*******************************************

**Comment by sv1eia** [comment]
@BuddyZhang1 

Did another new clean git clone repo
with up to latest commit 
-> [commit c47ff80](https://github.com/OrangePiLibra/OrangePi_H5SDK/commit/c47ff80cefec5cbe07b46d066a7a3d02f1e04268)

Unfortunately same exact errors remain
In trying to build a full xenial image got error

******************************
Conver script to dts ok.
update scp
pack boot package
GetPrivateProfileSection read to end
content_count=3
file /home/user/orangepi/OrangePi_H5SDK/output/pack/out/u-boot.fex cant be open
create package failed
ERROR: dragonsecboot  boot_package.cfg failed
******************************


And in trying to build only the uboot, got error

******************************
Compile U-boot......
Configuring for sun50iw2p1 - Board: sun50iw2p1, Options: SUN50IW2P1
Makefile:1203: *** missing separator.  Stop.
******************************

@BuddyZhang1 
Buddy, please, try to see yourself on a clean installation what is missing and add it.
Lately, you obviously added something in your working environment that is not included in the SDK and nothing can be built anymore from us that we use your SDK, either  package or a script parameter/configuration. So the best approach IMHO is to try yourself on a new clean development environment, try a git clone of this SDK and see what is missing.


I am using a Ubuntu Xenial 16.04.1 as the most appropriate build environment, I guess that is needed and everyone else is using it to build images.

**Comment by BuddyZhang1** [comment]
@sv1eia 
 Hi, firends,
 .... Pls try some modify...  modify .vimrc under current user
    set tabstop=4

**Comment by sv1eia** [comment]
@BuddyZhang1 

At line 1203 of file u-boot/Makefile YOU HAVE AN ERROR
Please, go there and you will see that the line is broken, its not a matter of tabstops.
You currently have there
***********************************
FINDFLAGS := -L 
-R --c++-kinds=+p --fields=+iaS --extra=+q
***********************************

whereas it should be
***********************************
FINDFLAGS := -L  -R --c++-kinds=+p --fields=+iaS --extra=+q
***********************************

IF you tried to build this yourself you would have seen it.

Are you even trying to build from your own SDK?

**Comment by sv1eia** [comment]
@BuddyZhang1 

Even if the above u-boot/Makefile gets corrected, the build fails later on at modules installation

********************************
 Build Mali450 succeed 
 Start Install Module 
make: Entering directory '/home/user/orangepi/OrangePi_H5SDK/kernel'
mkdir: cannot create directory ‘/home/user/orangepi/OrangePi_H5SDK/output/lib’: Not a directory
Makefile:970: recipe for target '_modinst_' failed
make: *** [_modinst_] Error 1
make: Leaving directory '/home/user/orangepi/OrangePi_H5SDK/kernel'
********************************

That error happened & reported many times before

**Comment by tkggithub** [comment]
Hi @BuddyZhang1,
I have opened issue#11 that has some more information about more specific u-boot build issues and some possible fixes.

**Comment by sv1eia** [comment]
@BuddyZhang1 

Tried another test.
Did a new clean git clone of the SDK repository. 
Up to [commit 136a9a7](https://github.com/OrangePiLibra/OrangePi_H5SDK/commit/136a9a753327899549bc28f5784cbac7f5e973f1)
Then tried to do a full release image build with xenial option.
System started to build and ended again with

***********************************
Conver script to dts ok.
update scp
pack boot package
GetPrivateProfileSection read to end
content_count=3
file /home/user/orangepi/OrangePi_H5SDK/output/pack/out/u-boot.fex cant be open
create package failed
ERROR: dragonsecboot  boot_package.cfg failed
***********************************


After some examination, the file u-boot.fex, does not actually exist.

It is generated from the build kernel option, so you have to take this into consideration and modify your build scripts so kernel build to be run prior to this stage.

After running the kernel build option, it gets generated and the above message is no longer displayed, but you need to configure better the build scripts in order to get this error fixed.

After that, all build options stop at the following error which has been reported many times (Release Image, Rootfs, Linux, Module build options fail)

***********************************
 Build Mali450 succeed 
 Start Install Module 
make: Entering directory '/home/user/orangepi/OrangePi_H5SDK/kernel'
mkdir: cannot create directory ‘/home/user/orangepi/OrangePi_H5SDK/output/lib’: Not a directory
Makefile:970: recipe for target '_modinst_' failed
make: *** [_modinst_] Error 1
make: Leaving directory '/home/user/orangepi/OrangePi_H5SDK/kernel'
***********************************

So, from the build options, only 'uboot' and 'kernel only' work, (without kernel modules, kernel only is useless), build release image option is still not working thus no image can be generated from the SDK.
Please have a look at these.

**Comment by sv1eia** [comment]
@BuddyZhang1 
New test results
Performed another clean git clone up to [commit dcf55745](https://github.com/OrangePiLibra/OrangePi_H5SDK/commit/dcf55745bedeec9d84fbaf6dd3c8fc9f4bfd4087)

On the new clean system, when selected the build release image, it fails with the many times reported in the past, "u-boot.fex cant be open" and build stops.
If I select the build uboot option it builds ok the uboot and then if I select (again) the build release image now it builds it ok.
So, that "u-boot.fex cant be open" needs to be fixed when a clean system is used.

@BuddyZhang1 
Upon trying to select the the ""Install Image into SDcard" option, it is asking the sudo password and the device to write but then it finishes immediately without any actual write on the SD. This needs to be fixed too.

thank you for your efforts!
